pkg/history/sqlitestore/store_test.go

```diff
@@ -0,0 +1,470 @@
+	}
+
+	expect := []Record{
+		{Source: source, CreateTime: originTime.Add(-1 * time.Hour), Payload: []byte("old-payload-3")},
+		{Source: source, CreateTime: originTime, Payload: []byte("new-payload")},
+	}
+	verifyRecords(t, db, ctx, expect)
+}
+
+func TestDatabase_InsertBulk_WithEarliestTime(t *testing.T) {
+	db := newTestDB(t)
+	ctx := context.Background()
+
+	now := time.Now()
+	source := "time-trim-source"
+	records := []Record{
+		{Source: source, CreateTime: now.Add(-3 * time.Hour), Payload: []byte("old-payload-1")},
+		{Source: source, CreateTime: now.Add(-2 * time.Hour), Payload: []byte("old-payload-2")},
+		{Source: source, CreateTime: now.Add(-1 * time.Hour), Payload: []byte("old-payload-3")},
+		{Source: source, CreateTime: now, Payload: []byte("new-payload")},
+	}
+
+	trimBefore := now.Add(-90 * time.Minute) // Trim records older than 90 minutes
+
+	err := db.InsertBulk(ctx, records, WithEarliestTime(trimBefore))
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	count, err := db.Count(ctx, source, 0, 0)
+	if err != nil {
+		t.Fatalf("unexpected error counting records: %v", err)
+	}
+
+	if count != 2 {
+		t.Errorf("expected 2 records after time-based trim, got %d", count)
+	}
+}
+
+func TestDatabase_Read(t *testing.T) {
+	db := newTestDB(t)
+	ctx := context.Background()
+
+	source := "test-source"
+	payload := []byte("test-payload")
+	at := time.Now()
+
+	_, err := db.Insert(ctx, Record{
+		Source:     source,
+		CreateTime: at,
+		Payload:    payload,
+	})
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Using zero-value Records to read the entire dataset
+	from := RecordID(0) // zero value = beginning of dataset
+	to := RecordID(0)   // zero value = end of dataset
+	into := make([]Record, 1)
+
+	count, err := db.Read(ctx, source, from, to, false, into)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	if count != 1 {
+		t.Errorf("expected 1 record, got %d", count)
+	}
+}
+
+func TestDatabase_Count(t *testing.T) {
+	db := newTestDB(t)
+	ctx := context.Background()
+
+	source := "test-source"
+	payload := []byte("test-payload")
+	at := time.Now()
+
+	_, err := db.Insert(ctx, Record{
+		Source:     source,
+		CreateTime: at,
+		Payload:    payload,
+	})
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Using zero-value Records to count the entire dataset
+	count, err := db.Count(ctx, source, 0, 0)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	if count != 1 {
+		t.Errorf("expected 1 record, got %d", count)
+	}
+}
+
+func TestDatabase_Size(t *testing.T) {
+	db := newTestDB(t)
+	ctx := context.Background()
+
+	size, err := db.Size(ctx)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	if size <= 0 {
+		t.Errorf("expected database size to be greater than 0, got %d", size)
+	}
+}
+
+func TestDatabase_LargeScale(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping large scale test in short mode")
+	}
```

Comment:
For this kind of test I think it'd be better to look for an env var (or something) instead of using `-short`. This isn't really testing anything, it's more about discovery and manual checking of performance.